test/unit/BPool/BPool_Bind_Unbind.tree

```diff
@@ -0,0 +1,51 @@
+BPool::bind
+├── when reentrancy lock is NOT set
+│   ├── when called NOT by controller
+│   │   └── it should revert
+│   └── when called by controller
+│       ├── when pool is finalized
+│       │   └── it should revert
+│       ├── when token is already bound
+│       │   └── it should revert
+│       ├── when MAX_BOUND_TOKENS tokens are already bound
+│       │   └── it should revert
+│       ├── when token weight is too low
+│       │   └── it should revert
+│       ├── when token weight is too high
+│       │   └── it should revert
+│       ├── when too little balance is provided
+│       │   └── it should revert
+│       ├── when weight sum exceeds MAX_TOTAL_WEIGHT
+│       │   └── it should revert
+│       └── when it is NOT finalized
+│           ├── it sets reentrancy lock
+│           ├── it increments _totalWeight
+│           ├── it calls _pullUnderlying
+│           ├── it adds token to the list
+│           └── it sets the token record
+└── when reentrancy lock is set
+    └── it should revert
+
+BPool::unbind
+├── when reentrancy lock is NOT set
+│   ├── when called NOT by controller
+│   │   └── it should revert
+│   └── when called by controller
+│       ├── when token is not bound
+│       │   └── it should revert
+│       └── when token is bound
+│           ├── when pool is finalized
+│           │   └── it should revert
+│           ├── when token is last on the tokens array
+│           │   ├── it sets reentrancy lock
+│           │   ├── it emits LOG_CALL event
+│           │   ├── it calls _pushUnderlying
+│           │   ├── it removes the token record
+│           │   ├── it decreases the total weight
+│           │   └── it pops from the array
+│           └── when token is NOT last one in the array
+│               ├── it swaps token with last one
+│               ├── it updates the swapped record index
+│               └── it pops from the array
```

Comment:
too code-related? i agree there are 2 branches (when is the last one and when it isn't) but perhaps it can be explained more humanly